internal/provider/resource_hvn_route.go

```diff
@@ -0,0 +1,71 @@
+package provider
+
+import (
+	"errors"
+	"time"
+
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var hvnRouteDefaultTimeout = time.Minute * 1
+
+func setHVNRouteResourceData(d *schema.ResourceData, route *networkmodels.HashicorpCloudNetwork20200907HVNRoute,
+	loc *sharedmodels.HashicorpCloudLocationLocation) error {
+
+	// Set self_link for the HVN route.
+	link := newLink(loc, HVNRouteResourceType, route.ID)
+	selfLink, err := linkURL(link)
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("self_link", selfLink); err != nil {
+		return err
+	}
+
+	// Set self_link identifying the target of the HVN route.
+	var targetLink string
+
+	switch route.Target.HvnConnection.Type {
```

Comment:
I think you can just use this type to build url, not necessary to have cases for everything

Reply:
i was thinking we need to handle the case when target has an unknown type but you're right, that's unnecessary since (a) unlikely to happen and (b) will not raise an error since it is still just a string